component/loki/source/file/tailer.go

```diff
@@ -61,6 +64,14 @@ func newTailer(metrics *metrics, logger log.Logger, handler loki.EntryHandler, p
 		positions.Remove(path, labels)
 	}
 
+	// If no cached position is found and the startFromEOF option is enabled.
+	if pos == 0 && startFromEOF {
+		pos, err = getLastLinePosition(path)
```

Comment:
Do you think we should also write the offset in the positions file? It will be written after the SyncPeriod ticker fires (every 10s), but I think it's a good idea to update it here as well.

Reply:
This way we would avoid recomputing the offset and we would read the logs which were added since the previous offset computation. I think that this is a minor improvement, but this is easy to add and if the user would rather start fresh from the end he can always delete the position